### ISSUE TITLE
warnings.warn(f"Subprocess {self.name} failed to start.")

### ISSUE THREAD

**Issue by Anonym0us33** [MineDojo/Voyager#9]
as title.
perplexity and gpt3 can't help with this
`
\Voyager\venv_Voyager\lib\site-packages\requests\adapters.py", line 519, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPConnectionPool(host='127.0.0.1', port=3000): Max retries exceeded with url: /start (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000016038136400>: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it'))
`
code:
```
from voyager import Voyager

voyager = Voyager(
    # azure_login=azure_login,
    mc_port = 65243,
    openai_api_key=openai_api_key
    # action_agent_model_name = 'gpt-3.5-turbo',
    # curriculum_agent_model_name = 'gpt-3.5-turbo'

    # curriculum_agent_qa_model_name = 'gpt-3.5-turbo',
    # critic_agent_model_name = 'gpt-3.5-turbo',
    # skill_manager_model_name = 'gpt-3.5-turbo',
)
voyager.learn()
```
steps:
install minecraft
find login from 2013
call microsoft
get told its out of hours
figure it out
install wrong version of fabric
install mods
forget to install final mod
compile it
download the jar version because you didnt read the instructions 
ask gpt
gpt figures out that its the wrong version of the mod because it read the tl;dr instructions
copy the compiled file
install another wrong version of fabric
install right version of fabric
follow minecraft instructions.
get port number
input it to script with API key
run the script as above

tl;dr STEPS TO REPRODUCE
`====================`
follow minecraft instructions to compile and install mods
get port number from game
input it to script with API key
run the script w/ 2 parameters as above
`voyager.learn()`
mineflayer crashes.
`====================`

**Comment by xieleo5** [comment]
Hi, thanks for your interest in this project. Could you please also provide the Mineflayer log? You can find it in the `logs` folder under your running directory.

**Comment by Anonym0us33** [comment]
Sure!
```
2023-05-28 03:14:51,818 - mineflayer - INFO - Stopping subprocess.
2023-05-28 03:14:52,828 - mineflayer - INFO - Starting subprocess with commands: ['node', '\Voyager\\voyager\\env\\mineflayer/index.js', '3000']
2023-05-28 03:14:52,881 - mineflayer - INFO - node:internal/modules/cjs/loader:1042
2023-05-28 03:14:52,881 - mineflayer - INFO - throw err;
2023-05-28 03:14:52,882 - mineflayer - INFO - ^
2023-05-28 03:14:52,881 - mineflayer - INFO - node:internal/modules/cjs/loader:1042
2023-05-28 03:14:52,881 - mineflayer - INFO - throw err;
2023-05-28 03:14:52,882 - mineflayer - INFO - ^
2023-05-28 03:14:52,882 - mineflayer - INFO - 
2023-05-28 03:14:52,882 - mineflayer - INFO - Error: Cannot find module 'express'
2023-05-28 03:14:52,882 - mineflayer - INFO - Require stack:
...


2023-05-28 03:15:03,178 - mineflayer - INFO - code: 'MODULE_NOT_FOUND',
2023-05-28 03:15:03,178 - mineflayer - INFO - requireStack: [ '\Voyager\\voyager\\env\\mineflayer\\index.js' ]
2023-05-28 03:15:03,179 - mineflayer - INFO - }
2023-05-28 03:15:03,179 - mineflayer - INFO - 
2023-05-28 03:15:03,179 - mineflayer - INFO - Node.js v18.13.0
```
heres the start and end. the full log is at https://pastebin.com/8xNryKs0
and another earlier log that is slightly different at
https://pastebin.com/NbKw5zTP
the difference is the first one starts multiple PIDs where the second only starts 2 or 3 in total

**Comment by xieleo5** [comment]
The issue will be closed due to a lack of activity, feel free to reopen it if you would like to continue the discussion.
